packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx

```diff
@@ -105,13 +105,13 @@ export class NavBase extends React.Component<INavProps, INavState> implements IN
 
   private _renderNavLink(link: INavLink, linkIndex: number, nestingLevel: number): JSX.Element {
     const { styles, groups, theme, onRenderLink = this._onRenderLink, linkAs: LinkAs = ActionButton } = this.props;
-
+    const isLinkWithIcon = link.iconProps && (!!link.iconProps.iconName || !!link.iconProps.imageProps);
```

Comment:
It should be sufficient to just check for `link.iconProps`. Otherwise if the icon interface ever changes this will have to be updated as well.